src/main/groovy/life/qbic/samplestatus/reporter/services/users/database/DatabaseSession.groovy

```diff
@@ -0,0 +1,97 @@
+package life.qbic.samplestatus.reporter.services.users.database
+
+import org.apache.commons.dbcp2.BasicDataSource
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import javax.annotation.PostConstruct
+import java.sql.Connection
+import java.sql.SQLException
+
+/**
+ * Creates a connection to the user database
+ *
+ * A class for setting up the connection to the user database. It should be used when data needs to be retrieved from the
+ * DB or written into it.
+ *
+ * @since: 1.0.0
+ * @author: Jennifer Bödker
+ *
+ */
+@Singleton(lazy = true)
+@Component
+class DatabaseSession implements ConnectionProvider {
+
+    private static DatabaseSession INSTANCE
+
+    private BasicDataSource dataSource
+
+    @Autowired
+    private UserDatabaseConfig userDatabaseConfig
```

Comment:
I think the abstraction is a little bit overkill. I dont see, where we have different config implementations.
Imo it is sufficient to ingest the configuration values directly in the service implementation (constructor ingestion)